### PR TITLE
Add SSLClusterConnection for connecting over TLS/SSL to Redis Cluster

### PR DESCRIPTION
**Pull request by dkent**

### CODE REVIEW

**Comment by Grokzen** [comment]
@dkent Looks good, straight forward and simple.

One comment is to add yourself to authors list in docs/ if you like or not.

I will have to read up on TLS/SSL certs and redis and give this a spin before merging but i see no real problems anyway :) When tested i will merge this.

**Comment by Grokzen** [comment]
@dkent All 3.2 redis builds have problems on travis. Will have to wait until fixed before travis-ci passes. All 3.0 tests works tho.

File: CHANGES [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Next release
```
Comment (Grokzen) [inline CHANGES]: @dkent Noticed one thing. you should not add this line to this file, you should add it to docs/release-notes.rst instead. Please move it there before i can merge this.

**Comment by Grokzen** [comment]
@dkent Except from the 1 comment in the code there is only one thing left to do and that is if you want add yourself to authors file. If not then please look at the comment and after that one is fixed i am ready to merge this.

File: CHANGES (hunk 1 above)
Comment (dkent) [inline CHANGES]: Hopefully I made the change correctly. Thanks! :)

**Comment by Grokzen** [comment]
@dkent There is some conflicts that you need to resolve now -_- you probably need to rebase on latest unstable

File: rediscluster/connection.py [hunk 2]
```diff
@@ -92,13 +121,15 @@ def __init__(self, startup_nodes=None, init_slot_cache=True, connection_class=Cl
...
+        if connection_class == SSLClusterConnection:
+            connection_kwargs['ssl'] = True  # needed in StrictRedis init
+
         self.nodes = NodeManager(
             startup_nodes,
             reinitialize_steps=reinitialize_steps,
             skip_full_coverage_check=skip_full_coverage_check,
             max_connections=self.max_connections,
             **connection_kwargs
-        )
```
Comment (dkent) [inline rediscluster/connection.py]: Looks like I made a mistake in the review screen. Fixing.

**Comment by Grokzen** [comment]
@dkent Ye i was pushing merges and changes myself so thats why your stuff broke :) sry for that

**Comment by dkent** [comment]
@Grokzen No worries. I was overly optimistic. I generally fetch and rebase.

**Comment by Grokzen** [comment]
@dkent sry but i had to merge this manually becuase there was issues with the merge of unstable you did into you branch. You can find the 2 relevant commits here dabfb97 && 88d38b5

Thanks for taking the time to implement this and submit the patch :)

Closing PR as fixed.

**Pull request closed without merging**